core/src/main/java/org/apache/druid/data/input/impl/FastLineIterator.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.data.input.impl;
+
+import com.google.common.base.Preconditions;
+import it.unimi.dsi.fastutil.bytes.ByteArrayList;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.java.util.common.parsers.CloseableIterator;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.NoSuchElementException;
+
+/**
+ * Like the Apache Commons LineIterator, but faster.
+ */
+public class FastLineIterator implements CloseableIterator<String>
+{
+  static final int BUFFER_SIZE = 512;
```

Comment:
All the static and instance variable fields should ideally be private unless there is a strong reason you want to relax their visibility.

Reply:
+1 for `CR`/`LF`.

For the buffer_size .. I use it in [tests such as this](https://github.com/apache/druid/pull/13545/files#diff-0a40f7c3d6223b3ada24ab09e70eb3f61c17728671de580dd96f22ea36d13ac5R171-R173), 

I can make it private but then we need to manually copy the value. Thoughts?

```
    // random lengths that force multiple buffer trips
    String line1 = RandomStringUtils.random(FastLineIterator.BUFFER_SIZE * 3 + 10);
    String line2 = RandomStringUtils.random(FastLineIterator.BUFFER_SIZE * 2 + 15);
    String line3 = RandomStringUtils.random(FastLineIterator.BUFFER_SIZE + 9);
```

Reply:
Sorry, I should have been more clear. I was referring to all the variables. 
```
static final byte CR = (byte) '\r';
static final byte LF = (byte) '\n';

final InputStream source;
final ByteArrayList buffer;

boolean finished;
String nextLine;
```

It is fine for constants like `BUFFER_SIZE` etc as non-private especially if you need to access them in tests.